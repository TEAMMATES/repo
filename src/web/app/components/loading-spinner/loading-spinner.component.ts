import { Component } from '@angular/core';

/**
 * Loading spinner to show when waiting for request.
 */
@Component({
  selector: 'tm-loading-spinner',
  templateUrl: './loading-spinner.component.html',
  styleUrls: ['./loading-spinner.component.scss'],
})
export class LoadingSpinnerComponent {
  // tmIsLoading: Number = 0;
  // initialTime = new Date().getMilliseconds;
  // endTime = new Date().getMilliseconds;
  // msg = '';

  // messageLoading(): void {
  //   setTimeout(() => { this.msg = 'Please wait, it is loading'; }, 10000);
  // }

}
