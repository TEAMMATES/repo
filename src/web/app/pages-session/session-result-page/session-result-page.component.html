<h1>
  Feedback Session Results
</h1>
<div class="card card-plain">
  <div class="card-body">
    <br/>
    <div class="row text-center">
      <div class="col-md-2 text-md-right"><b>Course ID:</b></div>
      <div class="col-md-5 text-md-left">{{ session.courseId }}</div>
    </div>
    <br/>
    <div class="row text-center">
      <div class="col-md-2 text-md-right"><b>Session:</b></div>
      <div class="col-md-5 text-md-left">{{ session.feedbackSessionName }}</div>
    </div>
    <br/>
    <div class="row text-center">
      <div class="col-md-2 text-md-right"><b>Opening time:</b></div>
      <div class="col-md-5 text-md-left">{{ formattedSessionOpeningTime }}</div>
    </div>
    <br/>
    <div class="row text-center">
      <div class="col-md-2 text-md-right"><b>Closing time:</b></div>
      <div class="col-md-5 text-md-left">{{ formattedSessionClosingTime }}</div>
    </div>
    <br/>
  </div>
</div>

<div class="card bg-light top-padded" *ngFor="let question of questions">
  <div class="card-body" style="padding: 20px;">
    <tm-question-text-with-info [questionNumber]="question.feedbackQuestion.questionNumber" [questionDetails]="question.feedbackQuestion.questionDetails" style="font-size: 20px;"></tm-question-text-with-info>
    <tm-student-view-responses *ngIf="question.responsesToSelf.length" [responses]="question.responsesToSelf" [questionDetails]="question.feedbackQuestion.questionDetails"></tm-student-view-responses>
    <div *ngFor="let responsesForOtherRecipient of question.otherResponses">
      <tm-student-view-responses *ngIf="responsesForOtherRecipient.length" [responses]="responsesForOtherRecipient" [questionDetails]="question.feedbackQuestion.questionDetails"></tm-student-view-responses>
    </div>
    <div *ngIf="question.responsesFromSelf.length" style="margin-top: 10px;">
      <strong>Your own responses:</strong>
      <div *ngFor="let responseFromSelf of question.responsesFromSelf">
        <tm-student-view-responses [responses]="[responseFromSelf]" [isSelfResponses]="true" [questionDetails]="question.feedbackQuestion.questionDetails"></tm-student-view-responses>
      </div>
    </div>
  </div>
</div>
