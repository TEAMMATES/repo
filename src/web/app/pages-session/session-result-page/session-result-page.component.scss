.card-plain {
  background-color: #EAEFF5;
  background-image: none;
  border: 0;
  box-shadow: 0 0 0 0;
}

.question-text {
  font-size: .9375rem;
}

.other-response {
  margin-top: 15px;
  margin-bottom: 25px;
}
