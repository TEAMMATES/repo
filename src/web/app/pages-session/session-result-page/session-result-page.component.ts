import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import moment from 'moment-timezone';
import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { TimezoneService } from '../../../services/timezone.service';
import {
  FeedbackSession, FeedbackSessionPublishStatus, FeedbackSessionSubmissionStatus,
  QuestionOutput,
  ResponseVisibleSetting,
  SessionResults,
  SessionVisibleSetting,
} from '../../../types/api-output';
import { Intent } from '../../../types/api-request';
import { ErrorMessageOutput } from '../../error-message-output';

/**
 * Feedback session result page.
 */
@Component({
  selector: 'tm-session-result-page',
  templateUrl: './session-result-page.component.html',
  styleUrls: ['./session-result-page.component.scss'],
})
export class SessionResultPageComponent implements OnInit {

  session: FeedbackSession = {
    courseId: '',
    timeZone: '',
    feedbackSessionName: '',
    instructions: '',
    submissionStartTimestamp: 0,
    submissionEndTimestamp: 0,
    gracePeriod: 0,
    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
    isClosingEmailEnabled: true,
    isPublishedEmailEnabled: true,
    createdAtTimestamp: 0,
  };
  questions: QuestionOutput[] = [];
  formattedSessionOpeningTime: string = '';
  formattedSessionClosingTime: string = '';

  constructor(private feedbackSessionsService: FeedbackSessionsService, private route: ActivatedRoute,
      private timezoneService: TimezoneService, private statusMessageService: StatusMessageService) {
    this.timezoneService.getTzVersion(); // import timezone service to load timezone data
  }

  ngOnInit(): void {
    this.route.queryParams.subscribe((queryParams: any) => {
      const { courseid: courseId, fsname: feedbackSessionName }: Record<string, string> = queryParams;
      this.feedbackSessionsService.getFeedbackSession({
        courseId,
        feedbackSessionName,
        intent: Intent.STUDENT_RESULT,
      }).subscribe((feedbackSession: FeedbackSession) => {
        const TIME_FORMAT: string = 'ddd, DD MMM, YYYY, hh:mm A zz';
        this.session = feedbackSession;
        this.formattedSessionOpeningTime =
            moment(this.session.submissionStartTimestamp).tz(this.session.timeZone).format(TIME_FORMAT);
        this.formattedSessionClosingTime =
              moment(this.session.submissionEndTimestamp).tz(this.session.timeZone).format(TIME_FORMAT);
        this.feedbackSessionsService.getFeedbackSessionsResult({
          courseId,
          feedbackSessionName,
          intent: Intent.STUDENT_RESULT,
        }).subscribe((sessionResults: SessionResults) => {
          this.questions = sessionResults.questions.sort(
              (a: QuestionOutput, b: QuestionOutput) =>
                  a.feedbackQuestion.questionNumber - b.feedbackQuestion.questionNumber);
        }, (resp: ErrorMessageOutput) => {
          this.statusMessageService.showErrorMessage(resp.error.message);
        });
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorMessage(resp.error.message);
      });
    });
  }

}
