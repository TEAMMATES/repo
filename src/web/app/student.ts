/**
 * A student.
 */
export interface Student {
  name: string;
  lastName: string;
}

/**
 * A student's response status for a feedback session.
 */
export interface StudentResponseStatus {
  email: string;
  name: string;
  sectionName: string;
  teamName: string;
  responseStatus: boolean;
}

/**
 * All the students' response status for a feedback session.
 */
export interface StudentsResponseStatus {
  studentsResponseStatus: StudentResponseStatus[];
}
