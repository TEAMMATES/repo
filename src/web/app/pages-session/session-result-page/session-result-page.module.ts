import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TimeDisplayerModule } from '../../components/time-displayer/time-displayer.module';
import { SessionResultPageComponent } from './session-result-page.component';

/**
 * Module for feedback session result page.
 */
@NgModule({
  imports: [
    CommonModule,
    TimeDisplayerModule,
  ],
  declarations: [
    SessionResultPageComponent,
  ],
  exports: [
    SessionResultPageComponent,
  ],
})
export class SessionResultPageModule { }
