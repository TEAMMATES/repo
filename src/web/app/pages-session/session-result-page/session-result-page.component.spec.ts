import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { of, throwError } from 'rxjs';
import { AuthService } from '../../../services/auth.service';
import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
import { NavigationService } from '../../../services/navigation.service';
import { StudentService } from '../../../services/student.service';
import {
  AuthInfo, FeedbackSession,
  FeedbackSessionPublishStatus,
  FeedbackSessionSubmissionStatus, RegkeyValidity,
  ResponseVisibleSetting,
  SessionVisibleSetting,
} from '../../../types/api-output';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { SingleStatisticsModule } from '../../components/question-responses/single-statistics/single-statistics.module';
import { StudentViewResponsesModule } from '../../components/question-responses/student-view-responses/student-view-responses.module';
import { QuestionTextWithInfoModule } from '../../components/question-text-with-info/question-text-with-info.module';
import { SessionResultPageComponent } from './session-result-page.component';
import Spy = jasmine.Spy;

describe('SessionResultPageComponent', () => {
  const testFeedbackSession: FeedbackSession = {
    feedbackSessionName: 'First Session',
    courseId: 'CS1231',
    timeZone: 'Asia/Singapore',
    instructions: '',
    submissionStartTimestamp: 0,
    submissionEndTimestamp: 1549095330000,
    gracePeriod: 0,
    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
    publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
    isClosingEmailEnabled: true,
    isPublishedEmailEnabled: true,
    createdAtTimestamp: 0,
  };
  const testInfo: AuthInfo = {
    masquerade: false,
    user: {
      id: 'user-id',
      isAdmin: false,
      isInstructor: true,
      isStudent: false,
    },
  };

  let component: SessionResultPageComponent;
  let fixture: ComponentFixture<SessionResultPageComponent>;
  let authService: AuthService;
  let navService: NavigationService;
  let studentService: StudentService;
  let feedbackSessionService: FeedbackSessionsService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        StudentViewResponsesModule,
        QuestionTextWithInfoModule,
        SingleStatisticsModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
      ],
      declarations: [SessionResultPageComponent],
      providers: [
        AuthService,
        NavigationService,
        StudentService,
        FeedbackSessionsService,
        {
          provide: ActivatedRoute,
          useValue: {
            queryParams: of({
              courseid: 'CS3281',
              fsname: 'Peer Feedback',
              key: 'reg-key',
            }),
          },
        },
      ],
    })
        .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SessionResultPageComponent);
    authService = TestBed.inject(AuthService);
    navService = TestBed.inject(NavigationService);
    studentService = TestBed.inject(StudentService);
    feedbackSessionService = TestBed.inject(FeedbackSessionsService);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with session results are loading', () => {
    component.isFeedbackSessionResultsLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when session results failed to load', () => {
    component.isFeedbackSessionResultsLoading = false;
    component.hasFeedbackSessionResultsLoadingFailed = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with user that is logged in and using session link', () => {
    component.regKey = 'session-link-key';
    component.loggedInUser = 'alice';
    component.personName = 'alice';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with user that is not logged in and using session link', () => {
    component.regKey = 'session-link-key';
    component.loggedInUser = '';
    component.personName = 'alice';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with an open feedback session with no questions', () => {
    component.session = {
      courseId: 'CS3281',
      timeZone: 'UTC',
      feedbackSessionName: 'Peer Review 1',
      instructions: '',
      submissionStartTimestamp: 1555232400,
      submissionEndTimestamp: 1555233400,
      gracePeriod: 0,
      sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
      responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
      submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
      publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
      isClosingEmailEnabled: true,
      isPublishedEmailEnabled: true,
      createdAtTimestamp: 1555231400,
    };
    component.questions = [];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should fetch auth info on init', () => {
    spyOn(authService, 'getAuthUser').and.returnValue(of(testInfo));

    component.ngOnInit();

    expect(component.courseId).toEqual('CS3281');
    expect(component.feedbackSessionName).toEqual('Peer Feedback');
    expect(component.regKey).toEqual('reg-key');
    expect(component.loggedInUser).toEqual('user-id');
  });

  it('should verify allowed access and used reg key', () => {
    const testValidity: RegkeyValidity = {
      isAllowedAccess: true,
      isUsed: true,
      isValid: false,
    };
    spyOn(authService, 'getAuthUser').and.returnValue(of(testInfo));
    spyOn(authService, 'getAuthRegkeyValidity').and.returnValue(of(testValidity));
    const navSpy: Spy = spyOn(navService, 'navigateByURLWithParamEncoding');

    component.ngOnInit();

    expect(navSpy.calls.count()).toEqual(1);
    expect(navSpy.calls.mostRecent().args[1])
        .toEqual('/web/student/sessions/result');
  });

  it('should load info for unused reg key that is allowed', () => {
    const testValidity: RegkeyValidity = {
      isAllowedAccess: true,
      isUsed: false,
      isValid: false,
    };
    spyOn(authService, 'getAuthUser').and.returnValue(of(testInfo));
    spyOn(authService, 'getAuthRegkeyValidity').and.returnValue(of(testValidity));
    spyOn(studentService, 'getStudent').and.returnValue(of({ name: 'student-name' }));
    spyOn(feedbackSessionService, 'getFeedbackSession').and.returnValue(of(testFeedbackSession));

    component.ngOnInit();

    expect(component.personName).toEqual('student-name');
    expect(component.session.courseId).toEqual('CS1231');
  });

  it('should deny access for reg key not belonging to logged in user', () => {
    const testValidity: RegkeyValidity = {
      isAllowedAccess: false,
      isUsed: false,
      isValid: true,
    };
    spyOn(authService, 'getAuthUser').and.returnValue(of(testInfo));
    spyOn(authService, 'getAuthRegkeyValidity').and.returnValue(of(testValidity));
    const navSpy: Spy = spyOn(navService, 'navigateWithErrorMessage');

    component.ngOnInit();

    expect(navSpy.calls.count()).toEqual(1);
    expect(navSpy.calls.mostRecent().args[1]).toEqual('/web/front');
    expect(navSpy.calls.mostRecent().args[2]).toEqual('You are not authorized to view this page.');
  });

  it('should deny access for invalid reg key', () => {
    const testValidity: RegkeyValidity = {
      isAllowedAccess: false,
      isUsed: false,
      isValid: false,
    };
    spyOn(authService, 'getAuthUser').and.returnValue(of(testInfo));
    spyOn(authService, 'getAuthRegkeyValidity').and.returnValue(of(testValidity));
    const navSpy: Spy = spyOn(navService, 'navigateWithErrorMessage');

    component.ngOnInit();

    expect(navSpy.calls.count()).toEqual(1);
    expect(navSpy.calls.mostRecent().args[1]).toEqual('/web/front');
    expect(navSpy.calls.mostRecent().args[2])
        .toEqual('You are not authorized to view this page.');
  });

  it('should navigate away when error occurs', () => {
    spyOn(authService, 'getAuthUser').and.returnValue(throwError({
      error: { message: 'This is error' },
    }));
    const navSpy: Spy = spyOn(navService, 'navigateWithErrorMessage');

    fixture.detectChanges();
    component.ngOnInit();

    expect(navSpy.calls.count()).toBe(1);
    expect(navSpy.calls.mostRecent().args[1]).toEqual('/web/front');
    expect(navSpy.calls.mostRecent().args[2])
        .toEqual('You are not authorized to view this page.');
  });

  it('should navigate to join course when user click on join course link', () => {
    component.regKey = 'reg-key';
    component.loggedInUser = 'user';
    const navSpy: Spy = spyOn(navService, 'navigateByURL');

    fixture.detectChanges();

    const btn: any = fixture.debugElement.nativeElement
        .querySelector('#join-course-btn');
    btn.click();

    expect(navSpy.calls.count()).toBe(1);
    expect(navSpy.calls.mostRecent().args[1]).toEqual('/web/join');
  });
});
