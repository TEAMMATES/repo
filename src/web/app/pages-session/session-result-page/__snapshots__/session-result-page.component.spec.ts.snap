// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionResultPageComponent should snap when session results failed to load 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed={[Function Boolean]}
  isFeedbackSessionResultsLoading="false"
  loggedInUser=""
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName=""
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        
        <div>
           You are viewing feedback results as . You may submit feedback for sessions that are currently open and view results without logging in. To access other features you need to 
          <a
            href="#"
          >
            login using a Google account
          </a>
           (recommended). 
        </div>
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    <div
      class="text-center"
    >
      <p
        class="text-muted m-1"
      >
        Failed to load results
      </p>
      <button
        class="btn btn-primary"
      >
        <i
          aria-hidden="true"
          class="fas fa-redo"
        />
        Â Retry
      </button>
    </div>
    
  </tm-loading-retry>
</tm-session-result-page>
`;

exports[`SessionResultPageComponent should snap with an open feedback session with no questions 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed="false"
  isFeedbackSessionResultsLoading="false"
  loggedInUser=""
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName=""
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        
        <div>
           You are viewing feedback results as . You may submit feedback for sessions that are currently open and view results without logging in. To access other features you need to 
          <a
            href="#"
          >
            login using a Google account
          </a>
           (recommended). 
        </div>
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          CS3281
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          Peer Review 1
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry>
</tm-session-result-page>
`;

exports[`SessionResultPageComponent should snap with default fields 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed="false"
  isFeedbackSessionResultsLoading="false"
  loggedInUser=""
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName=""
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        
        <div>
           You are viewing feedback results as . You may submit feedback for sessions that are currently open and view results without logging in. To access other features you need to 
          <a
            href="#"
          >
            login using a Google account
          </a>
           (recommended). 
        </div>
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry>
</tm-session-result-page>
`;

exports[`SessionResultPageComponent should snap with session results are loading 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed="false"
  isFeedbackSessionResultsLoading={[Function Boolean]}
  loggedInUser=""
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName=""
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        
        <div>
           You are viewing feedback results as . You may submit feedback for sessions that are currently open and view results without logging in. To access other features you need to 
          <a
            href="#"
          >
            login using a Google account
          </a>
           (recommended). 
        </div>
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-session-result-page>
`;

exports[`SessionResultPageComponent should snap with user that is logged in and using session link 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed="false"
  isFeedbackSessionResultsLoading="false"
  loggedInUser={[Function String]}
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName={[Function String]}
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        <div>
           You are viewing feedback results as alice. If you wish to link your Google account (alice) with this user, 
          <a
            href="#"
            id="join-course-btn"
          >
            click here
          </a>
          . 
        </div>
        
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry>
</tm-session-result-page>
`;

exports[`SessionResultPageComponent should snap with user that is not logged in and using session link 1`] = `
<tm-session-result-page
  authService={[Function AuthService]}
  backendUrl={[Function String]}
  courseId={[Function String]}
  feedbackSessionName={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  formattedSessionClosingTime=""
  formattedSessionOpeningTime=""
  hasFeedbackSessionResultsLoadingFailed="false"
  isFeedbackSessionResultsLoading="false"
  loggedInUser=""
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  personName={[Function String]}
  questions={[Function Array]}
  regKey={[Function String]}
  retryAttempts={[Function Number]}
  route={[Function Object]}
  router={[Function Router]}
  session={[Function Object]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
     Feedback Session Results
  
  </h1><div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="alert alert-primary"
        role="alert"
      >
        
        
        <div>
           You are viewing feedback results as alice. You may submit feedback for sessions that are currently open and view results without logging in. To access other features you need to 
          <a
            href="#"
          >
            login using a Google account
          </a>
           (recommended). 
        </div>
      </div>
    </div>
  </div><div
    class="card card-plain mt-3"
  >
    <div
      class="card-body"
    >
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Course ID:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="course-id"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Session:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="session-name"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Opening time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="opening-time"
        >
          
        </div>
      </div>
      <br />
      <div
        class="row text-center"
      >
        <div
          class="col-md-2 text-md-right"
        >
          <b>
            Closing time:
          </b>
        </div>
        <div
          class="col-md-5 text-md-left"
          id="closing-time"
        >
          
        </div>
      </div>
      <br />
    </div>
  </div><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry>
</tm-session-result-page>
`;
